### DISCUSSION TITLE
Is Pulse 3.7.1 compatible with Pulse Pro 2.0.1?

### DISCUSSION THREAD

**Discussion by briancordanyoung** [kean/Pulse#186] [General]
I have excitedly been integrating Pulse in to my iOS project and a separate macOS project.  In both cases the package resolved to the latest `3.7.1`. I downloaded [Pulse Pro 2.0.1](https://github.com/kean/PulsePro/releases/download/2.0.1/pulse-pro-2.0.1.zip) from GitHub releases.

When running Pulse Pro, both the macOS app and the iOS app show my Mac listed under "Remote Logging" when Pulse Pro is running. When selected, the activity spinner just keeps spinning and Pulse Pro does not show either "device" in the sidebar.

My Mac is controlled by our IT department and might be configured in a way blocking the connection. I've checked the firewall and allowed incoming connections to Pulse Pro. 

1. Are these version compatible with each other?
2. Is there some other setting on my Mac I can check to allow Pulse Pro to connect to my apps?

**Comment by briancordanyoung** [comment]
duh!  OK.  I found the warning:
<img width="782" alt="Screenshot 2023-05-03 at 7 53 01 PM" src="https://user-images.githubusercontent.com/104926/236101845-25e227b3-a080-4d99-92ef-5fea57ae621f.png">

**Reply by kean** [reply]
> Are these version compatible with each other?

Yes, Pulse 3.x is designed to be compatible with Pulse Pro 2.x. The latest test I've done was with Pulse 3.7.0 and Pulse Pro 2.0.1.

Pulse for Mac is coming to App Store soon, hopefully within a couple of weeks. It's a massive improvement compared to Pulse Pro.

> Is there some other setting on my Mac I can check to allow Pulse Pro to connect to my apps?

Make sure to follow steps 4.1-4.2 from the [Getting Started](https://kean-docs.github.io/pulse/documentation/pulse/gettingstarted) guide.

It's a good start that it sees the Mac – it means the Bonjour is working. The service itself runs on the unspecified port ([`.any`](https://developer.apple.com/documentation/network/nwendpoint/port/2998599-any)). I myself haven't run into any issues with it. Pulse Pro can also be configured to use a specific port in case the network blocks some of the ports. You may want to try running it on one of the well-known ports.

If it still doesn't work, please collect some details about the error. The framework is fully open-source. To debug it, see `RemoteLogger-Connection.swift`.

**Reply by briancordanyoung** [reply]
Thank you for the quick response to a closed discussion! 😁

There is definitely something up with my machine.  I've followed the getting started steps closely. I've installed the Pulse for macOS  alpha and even tried syncing the package version with the same version.  In all cases Bonjour is working, and showing the Pulse for macOS from my apps—they just don't show as a device in the Pulse.app. I'll keep debugging. Thanks for some great software. Nuke is a favorite of mine.

**Reply by briancordanyoung** [reply]
> There is definitely something up with my machine.

And... I confirmed that there must be something about how IT has locked this Mac down.  I've installed both Pulse.app and my macOS app on my personal Mac and it all works together great. 🤔
